apps/chat/src/components/Marketplace/ApplicationDetails/ApplicationFooter.tsx

```diff
@@ -15,6 +27,60 @@ interface Props {
   onUseEntity: () => void;
 }
 
+const ApplicationPublishActions = ({ entity }: Pick<Props, 'entity'>) => {
+  const { t } = useTranslation(Translation.Chat);
+
+  const [publishAction, setPublishAction] = useState<PublishActions>();
+
+  const handleClosePublishModal = useCallback(() => {
+    setPublishAction(undefined);
+  }, []);
+
+  const publishEntity = {
+    name: entity.name,
+    id: ApiUtils.decodeApiUrl(entity.id),
+    folderId: getFolderIdFromEntityId(entity.id),
+  };
+
+  return (
+    <>
+      <Tooltip
+        tooltip={isItemPublic(entity.id) ? t('Unpublish') : t('Publish')}
+      >
+        <button
+          onClick={() =>
+            setPublishAction(
+              isItemPublic(entity.id)
+                ? PublishActions.DELETE
+                : PublishActions.ADD,
+            )
+          }
+          className="group flex size-[34px] items-center justify-center rounded text-secondary hover:bg-accent-primary-alpha hover:text-accent-primary"
+          data-qa="application-share"
+        >
+          {isItemPublic(entity.id) ? (
+            <UnpublishIcon className="size-6 shrink-0 cursor-pointer text-secondary hover:text-accent-primary group-hover:text-accent-primary" />
+          ) : (
+            <IconWorldShare
+              size={24}
+              className="shrink-0 cursor-pointer text-secondary group-hover:text-accent-primary"
+            />
+          )}
+        </button>
+      </Tooltip>
+      {publishAction && (
```

Comment:
Isn't it better to place PublishModal in the Marketplace component? That way we can reuse it for the cards as well.